app/index.js

```diff
@@ -35,24 +38,99 @@ import * as ReduxActions from './actions'; //Import your actions
 import styles from './styles';
 
 class Main extends Component {
+    constructor() {
+        super();
+        this.state = {key: '', iv: ''};
+        this.init();
+        this.getPaths = this.getPaths.bind(this);
+    }
+
+    //Create directory, files, and AES values
+    init(){
+        const dirs = RNFetchBlob.fs.dirs;
+        paths = this.getPaths();
+
+        RNFetchBlob.fs.mkdir(dirs.DocumentDir + '/idly/')
+        .catch((err) => {});
+
+        RNFetchBlob.fs.createFile(
+            paths.cardsPath,
+            '',
+            'utf8'
+        )
+        .catch((err) => {});
+
+        RNFetchBlob.fs.createFile(
+            paths.messagesPath,
+            '',
+            'utf8'
+        )
+        .catch((err) => {});
+
+        SInfo.getItem('key', {})
+        .then((value) => {
+            if (value === null){
+                for(var key = ''; key.length < 16;)
+                    key += Math.random().toString(36).substr(2, 1)
+                SInfo.setItem('key', key, {});
+                this.setState({key: key});
+            }
+            else
+                this.setState({key: value});
+        });
+
+        SInfo.getItem('iv', {})
+        .then((value) => {
+            if (value === null){
+                for(var iv = ''; iv.length < 16;)
+                    iv += Math.random().toString(36).substr(2, 1)
+                SInfo.setItem('iv', iv, {});
+                this.setState({iv: iv});
+            }
+            else
+                this.setState({iv: value});
+        });
+    }
+
+    getPaths(){
+        const dirs = RNFetchBlob.fs.dirs;
+        var cardsPath = '/idly/cards.dat';
+        var messagesPath = '/idly/messages.dat';
+        if (Platform.OS === 'ios') {
+            cardsPath = `${dirs.DocumentDir}${cardsPath}`;
+            messagesPath = `${dirs.DocumentDir}${messagesPath}`;
+        } else {
+            cardsPath = dirs.DocumentDir + cardsPath;
+            messagesPath = dirs.DocumentDir + messagesPath;
+        }
+        return {cardsPath: cardsPath, messagesPath: messagesPath};
+    }
+
     componentDidMount() {
+        var paths = this.getPaths();
         var _this = this;
         //Check if any card data exists
-        AsyncStorage.getItem('carddata', (err, carddata) => {
-            //if it doesn't exist, extract from json file
-            //save the initial data in Async
-            if (carddata === null){
-                AsyncStorage.setItem('carddata', JSON.stringify(CardData.card));
+        RNFetchBlob.fs.readFile(paths.cardsPath, 'utf8')
+        .then((carddata) => {
+            if (carddata === ''){
```

Comment:
2018-05-12 15:16:31.956 [info][tid:com.facebook.react.JavaScript] 'key: ', undefined
2018-05-12 15:16:31.956569-0700 idly[1559:30315] 'key: ', undefined
2018-05-12 15:16:31.957 [info][tid:com.facebook.react.JavaScript] 'iv: ', undefined
2018-05-12 15:16:31.957123-0700 idly[1559:30315] 'iv: ', undefined
JSON.stringify(CardData.card) prints the cards